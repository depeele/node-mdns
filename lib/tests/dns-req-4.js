var Util    = require('util'),
    Utils   = require('../utils.js'),
    Message = require('../message.js'),

    data    = [
        0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x00,
        0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x32,
        0x01, 0x32, 0x01, 0x30, 0x02, 0x31, 0x30,
        0x07, 0x69, 0x6e, 0x2d, 0x61, 0x64, 0x64,
        0x72, 0x04, 0x61, 0x72, 0x70, 0x61, 0x00,
        0x00, 0x0c, 0x00, 0x01
    ];

/*****************************************************************************
 * Unpack test
 *
 */
console.log("--------------------------------------------------------------");
var rBuf    = new Buffer( data ),
    message = new Message( rBuf );

console.log("Unpack mesage from %d bytes of raw data:", message.consumed);
console.log("message:\n%s", JSON.stringify(message, null, 2));

/*****************************************************************************
 * Pack test
 *
 */
console.log("--------------------------------------------------------------");
var wBuf    = new Buffer( 4096 );

message.pack(wBuf);

console.log("Packed message (%d bytes):\n%s",
            message.produced, Utils.buf2hex(wBuf, {
                                            length:     message.produced,
                                            octetsPer:  16
                             }));

if (message.produced !== data.length)
{
    console.log("*** Pack error: length %d != expected %d",
                message.produced, data.length);
    process.exit(-1);
}

var diff    = 0;
for (var idex = 0, len = data.length; idex < len; idex++)
{
    if (wBuf[idex] !== data[idex])
    {
        diff++;
        console.log("*** Pack error: @ %s: %s != expected %s",
                    Utils.int2hex(idex, 2),
                    Utils.int2hex(wBuf[idex]),
                    Utils.int2hex(data[idex]));

        if (diff > 8)   { break; }
    }
}

if (diff === 0)
{
    console.log(">>> packed buffer matches original source");
}
