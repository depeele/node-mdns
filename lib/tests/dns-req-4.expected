--------------------------------------------------------------
Unpack mesage from 39 bytes of raw data:
message:
{ header: 
   { id: 0,
     opcode: 0,
     qr: 0,
     aa: 0,
     tc: 0,
     rd: 0,
     ra: 0,
     z: 0,
     ad: 0,
     cd: 0,
     rcode: 0,
     qdCount: 1,
     anCount: 0,
     nsCount: 0,
     arCount: 0,
     consumed: 12 },
  question: 
   [ { qname: '2.2.0.10.in-addr.arpa.',
       qtype: 12,
       qclass: 1,
       consumed: 27 } ],
  answer: [],
  authority: [],
  additional: [],
  consumed: 39 }
--------------------------------------------------------------
Packed message (39 bytes):
 0x0000: 00 00 00 00 00 01 00 00 00 00 00 00 01 32 01 32 
 0x0010: 01 30 02 31 30 07 69 6e 2d 61 64 64 72 04 61 72 
 0x0020: 70 61 00 00 0c 00 01
>>> packed buffer matches original source
